packages/blade/src/components/Input/BaseInput/BaseInput.tsx

```diff
@@ -846,105 +902,118 @@ const _BaseInput: React.ForwardRefRenderFunction<BladeElementRef, BaseInputProps
+                autoFocus={autoFocus}
+                keyboardReturnKeyType={keyboardReturnKeyType}
+                keyboardType={keyboardType}
+                autoCompleteSuggestionType={autoCompleteSuggestionType}
+                accessibilityProps={accessibilityProps}
+                currentInteraction={currentInteraction}
+                setCurrentInteraction={setCurrentInteraction}
+                numberOfLines={numberOfLines}
+                isTextArea={isTextArea || maxTagRows === 'multiple' || maxTagRows === 'expandable'}
+                hasPopup={hasPopup}
+                hasTags={!!(tags && tags.length > 0)}
+                shouldIgnoreBlurAnimation={shouldIgnoreBlurAnimation}
+                autoCapitalize={autoCapitalize}
+                isDropdownTrigger={isDropdownTrigger}
+                $size={size}
+                {...metaAttribute({ name: MetaConstants.StyledBaseInput })}
+              />
+            </BaseInputTagSlot>
+            <BaseInputVisuals
               interactionElement={interactionElement}
               suffix={suffix}
               trailingIcon={trailingIcon}
-              maxCharacters={maxCharacters}
-              textAlign={textAlign}
-              // eslint-disable-next-line jsx-a11y/no-autofocus
-              autoFocus={autoFocus}
-              keyboardReturnKeyType={keyboardReturnKeyType}
-              keyboardType={keyboardType}
-              autoCompleteSuggestionType={autoCompleteSuggestionType}
-              accessibilityProps={accessibilityProps}
-              currentInteraction={currentInteraction}
-              setCurrentInteraction={setCurrentInteraction}
-              numberOfLines={numberOfLines}
-              isTextArea={isTextArea || maxTagRows === 'multiple' || maxTagRows === 'expandable'}
-              hasPopup={hasPopup}
-              hasTags={!!(tags && tags.length > 0)}
-              shouldIgnoreBlurAnimation={shouldIgnoreBlurAnimation}
-              autoCapitalize={autoCapitalize}
-              isDropdownTrigger={isDropdownTrigger}
-              {...metaAttribute({ name: MetaConstants.StyledBaseInput })}
+              isDisabled={isDisabled}
+              validationState={validationState}
+              trailingActionButton={trailingActionButton}
+              size={size}
             />
-          </BaseInputTagSlot>
-          <BaseInputVisuals
-            interactionElement={interactionElement}
-            suffix={suffix}
-            trailingIcon={trailingIcon}
-            isDisabled={isDisabled}
-          />
-        </BaseInputWrapper>
+          </BaseInputWrapper>
+        </FocusRingWrapper>
       </BaseBox>
-      {/* the magic number 136 is basically max-width of label i.e 120 and then right margin i.e 16 which is the spacing between label and input field */}
+      {/* the magic number 192 is basically max-width of label i.e 176 and then right margin i.e 16 which is the spacing between label and input field */}
```

Comment:
Can we move this to token file of baseinput? saw this being used in file upload also

Reply:
Done